### PR TITLE
RNRBC-72 Converted eithers to trys to improve information that can be…

### PR DESCRIPTION
**Pull request by adwelly**
… logged.

### CODE REVIEW

File: app/uk/gov/hmrc/residencenilratebandcalculator/JsonBuilder.scala [hunk 1]
```diff
@@ -42,32 +45,32 @@ class JsonBuilder @Inject()(rnrbConnector: RnrbConnector) {
...
-      case Left(error) => Left(error)
+      case Failure(exception) => Failure(exception)
     }
   }
 
-  def build(sessionConnector: SessionConnector)(implicit headerCarrier: HeaderCarrier): Future[Either[String, JsValue]] = {
+  def build(sessionConnector: SessionConnector)(implicit headerCarrier: HeaderCarrier) = {
     sessionConnector.fetch.flatMap {
-      case None => Future.successful(Left("No cache map returned"))
+      case None => Future.successful(Failure(new NoCacheMapException("No cache map.")))
```
Comment (HughPowell) [inline app/uk/gov/hmrc/residencenilratebandcalculator/JsonBuilder.scala]: Can we have a more meaningful error?  The type, NoCacheMapException, suggests there's no cache map.
"Unable to retrieve cache map from SessionConnector" maybe?

File: app/uk/gov/hmrc/residencenilratebandcalculator/connectors/RnrbConnector.scala [hunk 2]
```diff
@@ -50,11 +52,15 @@ class RnrbConnector @Inject()(http: WSHttp) extends ServicesConfig {
...
 
-  def getSuccessfulResponseSchema: Future[Either[String, SchemaType]] =
+  def getSuccessfulResponseSchema =
     http.GET(s"$serviceUrl${baseSegment}schemas/deceaseds-estate.jsonschema").map {
       response => Json.fromJson[SchemaType](response.json) match {
-        case JsSuccess(schema, _) => Right(schema)
-        case error: JsError => Left((JsError.toJson(error) \ "obj" \ 0 \ "msg" \ 0).as[String])
+        case JsSuccess(schema, _) => Success(schema)
+        case error: JsError => {
+          Left((JsError.toJson(error) \ "obj" \ 0 \ "msg" \ 0).as[String])
```
Comment (HughPowell) [inline app/uk/gov/hmrc/residencenilratebandcalculator/connectors/RnrbConnector.scala]: I think we want to remove this line?

File: app/uk/gov/hmrc/residencenilratebandcalculator/controllers/ResultsController.scala [hunk 3]
```diff
@@ -35,14 +35,16 @@ class ResultsController @Inject()(appConfig: FrontendAppConfig, val messagesApi:
   extends FrontendController with I18nSupport {
 
   def onPageLoad = Action.async { implicit request => {
-    val jsonEither: Future[Either[String, JsValue]] = jsonBuilder.build(sessionConnector)
+    val jsonEither = jsonBuilder.build(sessionConnector)
```
Comment (HughPowell) [inline app/uk/gov/hmrc/residencenilratebandcalculator/controllers/ResultsController.scala]: This is no longer a jsonEither, jsonTry maybe?

File: app/uk/gov/hmrc/residencenilratebandcalculator/json/JsonBuilder.scala [hunk 4]
```diff
@@ -42,32 +47,33 @@ class JsonBuilder @Inject()(rnrbConnector: RnrbConnector) {
...
     futureSchema.map {
-      case Right(schema) => {
+      case Success(schema) => {
         val incomingJson = Json.toJson(setKeys(cacheMap))
         val validationResult = validator.validate(schema, incomingJson).asEither
         validationResult match {
           case Left(error) => {
-            val errorString = error.seq.flatMap(_._2).map(_.message).foldLeft(new StringBuilder())(_ append _).toString()
-            Left(errorString)
+            //val errorString = error.seq.flatMap(_._2).map(_.message).foldLeft(new StringBuilder())(_ append _).toString()
```
Comment (TheNickWilson) [inline app/uk/gov/hmrc/residencenilratebandcalculator/json/JsonBuilder.scala]: Can we remove this commented code please?

File: app/uk/gov/hmrc/residencenilratebandcalculator/json/JsonErrorProcessor.scala [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.residencenilratebandcalculator.json
+
+import play.api.data.validation.ValidationError
+import play.api.libs.json.{JsPath, JsValue}
+
+/**
+  * Created by andy on 19/12/2016.
```
Comment (TheNickWilson) [inline app/uk/gov/hmrc/residencenilratebandcalculator/json/JsonErrorProcessor.scala]: Can we remove this comment (and turn its generation off)?

File: test/uk/gov/hmrc/residencenilratebandcalculator/json/JsonErrorProcessorSpec.scala [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+package uk.gov.hmrc.residencenilratebandcalculator.json
+
+import com.eclipsesource.schema.{SchemaType, SchemaValidator}
+import play.api.data.validation.ValidationError
+import play.api.libs.json._
+import uk.gov.hmrc.play.test.UnitSpec
+
+/**
+  * Created by andy on 19/12/2016.
```
Comment (TheNickWilson) [inline test/uk/gov/hmrc/residencenilratebandcalculator/json/JsonErrorProcessorSpec.scala]: Can we remove this comment (and turn its generation off)?

**Pull request merged**